app/src/main/kotlin/github/com/rhacco/dota2androidapp/viewmodel/MatchesViewModel.kt

```diff
@@ -36,20 +36,23 @@ class MatchesViewModel(application: Application) : AndroidViewModel(application)
                 ))
     }
 
+    fun clearLiveMatches() = TopLiveGamesRepository.clearTopLiveGames()
+
     fun getLiveMatchListData(averageMMR: Int, serverSteamId: Long) {
         mIsLoadingLiveData.value = true
         mDisposables.add(RealtimeStatsRepository.getRealtimeStats(serverSteamId)
                 .subscribe(
                         { result ->
                             mIsLoadingLiveData.value = false
-                            val listItemData: LiveMatchesListAdapter.ListItemData =
+                            val newItem: LiveMatchesListAdapter.ListItemData =
                                     LiveMatchesListAdapter.ListItemData()
-                            listItemData.mAverageMMR = averageMMR
+                            newItem.mAverageMMR = averageMMR
+                            newItem.mMatchID = result.match.matchid
                             if (averageMMR < 1)
-                                listItemData.mTitle = "Tournament Match (Match ID " +
+                                newItem.mTitle = "Tournament Match (Match ID " +
                                         result.match.matchid + ")"
                             else
-                                listItemData.mTitle = "Ranked Match (Average MMR " + averageMMR +
+                                newItem.mTitle = "Ranked Match (Average MMR " + averageMMR +
                                         ", Match ID " + result.match.matchid + ")"
                             if (result.teams.size == 2) {
                                 result.teams[0].players
```

Comment:
Don't repeat yourself. You have result.team[0] and result.team[1] that can easily be put into a loop (or stream)

Also, when is result.team.size not equal to 2?

Reply:
Already fixed in #9 

We are fetching data from an external source that might malfunction at some point. So I'd rather add a quick safety check now and some debug information to make debugging easier when needed